src/http/request.cr

```diff
@@ -11,7 +11,8 @@ class HTTP::Request
   @query_params : Params?
   @uri : URI?
 
-  def initialize(@method : String, @resource : String, @headers : Headers = Headers.new, @body = nil, @version = "HTTP/1.1")
+  def initialize(@method : String, @resource : String, headers : Headers? = nil, @body = nil, @version = "HTTP/1.1")
+    @headers = Headers.new_or_dup(headers)
```

Comment:
Mh, yeah, I initially had it in multiple places so it kind of made sense, but then could workaround doing a second copy so the other uses vanished.